web/libs/editor/src/components/InteractiveOverlays/CommentsOverlay.tsx

```diff
@@ -152,10 +210,15 @@ const CommentsOverlayInner: React.FC<CommentsOverlayProps> = observer(({ annotat
   return (
     // biome-ignore lint/a11y/noSvgWithoutTitle: It's not just an icon or a figure; it's an entire interactive layer.
     <svg className={containerStyles.join(" ")} ref={setRef} xmlns="http://www.w3.org/2000/svg">
-      {overlayComments.map((comment: MSTComment) => {
-        const { id } = comment;
-        return <CommentItem key={[id, uniqKey].join("-")} comment={comment} rootRef={rootRef} />;
-      })}
+      <g key={uniqKey}>
+        {annotation.isLinkingMode && annotation.results.map((result: MSTResult) => (
```

Comment:
Most probably we need to check exactly one specific linking mode here.
```jsx
{annotation.linkingMode === LINK_COMMENT_MODE && annotation.results.map((result: MSTResult) => (
```

Reply:
Yeah, you are right!
```suggestion
        {annotation.linkingMode === LINK_COMMENT_MODE && annotation.results.map((result: MSTResult) => (
```